CardPayments/src/main/java/com/paypal/android/cardpayments/CardClient.kt

```diff
@@ -65,6 +67,15 @@ class CardClient internal constructor(
     }
 
     private suspend fun confirmPaymentSource(activity: FragmentActivity, cardRequest: CardRequest) {
+        try {
+            cardAPI.fetchCachedOrRemoteClientID()
```

Comment:
q: should we fail if we can't get client id here? Confirm payment source may still be able to succeed with the access token.

Reply:
We decided in one of our group meetings w/ David and Walter that we want all flows to fail unless we have a clientID, even if the flow itself doesn't need a clientID.

Reply:
In follow up PRs, we want to fetch the clientID as soon as possible (when merchant gives ur their accessToken) similar to the config on BT.

Reply:
Oh ok got it. Would `fun prefetchClientID(): Unit` make sense? Or maybe it's OD.

Reply:
There is a flow where the value of `clientID` is actually needed within the flow (NXO) - [code](https://github.com/paypal/Android-SDK/blob/5e950e2d268fe7c1dd8bd9749b643b11458c38a2/PayPalNativePayments/src/main/java/com/paypal/android/paypalnativepayments/PayPalNativeCheckoutClient.kt#L67).

But yeh, the other 2 flows (card & PP web) ignore the return value in the flow, and only use it for analytics metadata.

Down to rename this if we think there is something more clear!

Reply:
Yeah I would just to make the intention clear. It makes sense now but as the code evolves it might not be clear that it's a prefetch. Maybe we could have two methods? Like `suspend fun prefetchClientID(): Unit` and `suspend fun fetchClientID(): String`? The caller won't care if it's from a cache or the network, we can do some information hiding to keep the api clean too.

Reply:
Yeh, that's a good call. There is no pre-fetch happening now. But when we do, we should make sure the naming is clear.

I added a note about this conversion in that follow-up [JIRA](https://paypal.atlassian.net/browse/DTNOR-823).